corehq/util/celery_utils.py

```diff
@@ -0,0 +1,149 @@
+import kombu.five
+from celery import Celery
+from django.conf import settings
+from datetime import datetime
+from time import sleep, time
+
+
+class TaskInfo(object):
+
+    def __init__(self, _id, name, time_start=None):
+        self.id = _id
+        self.name = name
+        # http://stackoverflow.com/questions/20091505/celery-task-with-a-time-start-attribute-in-1970
+        if time_start:
+            self.time_start = datetime.fromtimestamp(time() - kombu.five.monotonic() + time_start)
+        else:
+            self.time_start = None
+
+
+class InvalidTaskTypeError(Exception):
+    pass
+
+
+def get_active_task_info(tasks):
+    result = []
+    for task_dict in tasks:
+        result.append(TaskInfo(task_dict.get('id'), task_dict.get('name'), task_dict.get('time_start')))
+    return result
+
+
+def get_reserved_task_info(tasks):
+    result = []
+    for task_dict in tasks:
+        result.append(TaskInfo(task_dict.get('id'), task_dict.get('name')))
+    return result
+
+
+def get_scheduled_task_info(tasks):
+    result = []
+    for task_dict in tasks:
+        task_dict = task_dict.get('request', {})
```

Comment:
is this just some weird difference in how celery internals represents scheduled tasks?